src/submit_jobs.py

```diff
@@ -0,0 +1,116 @@
+import argparse
+from datetime import datetime
+
+from utils.helpers import (
+    construct_predicates,
+    load_job_config,
+    read_predicates,
+)
+from utils.spark import SharedSparkSession, SparkJob
+
+# Default values for jobs, used per job if not explicitly set in the job's
+# input JSON. CUR and YEAR values are only used if use_partitions is true in
+# the job definition. use_predicates should be disabled for any table without
+# a PARID column
+DEFAULT_VAR_CUR = ["Y", "N", "D"]
+DEFAULT_VAR_MIN_YEAR = 1999
+DEFAULT_VAR_MAX_YEAR = datetime.now().year
+DEFAULT_VAR_USE_PREDICATES = True
+DEFAULT_VAR_USE_PARTITIONS = True
+
+# Constants for paths inside the Spark container
+PATH_IPTS_PASSWORD = "/run/secrets/IPTS_PASSWORD"
+PATH_PREDICATES = "/tmp/config/default_predicates.csv"
+PATH_INITIAL_DIR = "/tmp/target/initial"
+PATH_FINAL_DIR = "/tmp/target/final"
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Submit iasWorld Spark extraction jobs"
+    )
+    parser.add_argument(
+        "--json-file",
+        type=str,
+        nargs="?",
+        help="Path to a JSON file containing job configuration(s)",
+    )
+    parser.add_argument(
+        "--json-string",
+        type=str,
+        nargs="?",
+        help="JSON string containing job configurations(s)",
+    )
+
+    return parser.parse_args()
+
+
+def main() -> str:
+    # Get the job definition(s) from the argument JSON
+    args = parse_args()
+    job_config = load_job_config(args)
+
+    # Predicates are used to slice up a table/job into discrete chunks to
+    # improve parallelization. In our case, we slice tables into equally-sized
+    # chunks by PARID (where available) since it is indexed (we get fast reads)
+    predicates_csv = read_predicates(PATH_PREDICATES)
+
+    # Each session is shared across all read jobs and manages job order,
+    # credentialing, retries, etc.
+    current_datetime = datetime.now().strftime("%Y-%m-%d_%H:%M:%S")
+    session_name = f"iasworld_{current_datetime}"
+    session = SharedSparkSession(
+        app_name=session_name, password_file_path=PATH_IPTS_PASSWORD
+    )
```

Comment:
Note that you can connect to an existing session by passing the same name, which might be helpful in a future where we want to poll the status of jobs.